misk/src/main/kotlin/misk/web/jetty/JettyService.kt

```diff
@@ -337,15 +336,34 @@ class JettyService @Inject internal constructor(
         // the address is a null byte ('\0'). The address has no connection with filesystem
         // path names.
       }
-      server.destroy()
+
+      logger.info { "Stopped Jetty in $stopwatch" }
     }
 
     if (healthExecutor != null) {
       healthExecutor!!.shutdown()
       healthExecutor!!.awaitTermination(10, TimeUnit.SECONDS)
     }
+  }
+
+  override fun shutDown() {
+    // We need jetty to shut down at the very end to keep outbound connections alive
+    // (due to sidecars). As such, we wait for `shutdown_sleep_ms` so that our
+    // in flight requests drain, but we don't shut down dependencies until after.
+    //
+    // The true jetty shutdown occurs in stop() above, called from MiskApplication.
+    //
+    // Ideally we could call jetty.awaitInflightRequests() but that's not available
+    // for us.
+    if (webConfig.shutdown_sleep_ms > 0) {
+      sleep(webConfig.shutdown_sleep_ms.toLong())
```

Comment:
Could we run into issues where we try to shut down "work producing" services and Jetty is in the middle of that list? e.g. we get a shutdown order like 

1. Cron
2. Jetty
3. SQS

wouldn't SQS's shutdown be delayed by `shutdown_sleep_ms` and happen after Jetty?

Reply:
yea, sqs shutdown could happen after the sleeping which is fine in my opinion

Reply:
i think the proper fix to that, if we think it's a problem, is to wait for kubernetes 1.28 or to shutdown many things in parallel when possible